[AOTI] Fix a cpp kernel missing arg type issue

desertfire opened the pull request:
Summary: The current way of fetching the kernel arg types only works for tensors, not symbols.

Test Plan: CI

Differential Revision: D53111282




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

ezyang commented on torch/_inductor/ir.py:
```diff
@@ -4396,6 +4396,7 @@ def should_allocate(self):
 
     # TODO: handle bools carefully
     def __init__(self, sym, data):
+        data.realize()
```
also in https://github.com/pytorch/pytorch/pull/118125

ezyang commented on torch/_inductor/codegen/wrapper.py:
```diff
@@ -3042,7 +3043,14 @@ def generate_default_grid(self, name: str, grid: List[Any], cuda: bool = True):
+        device_index=None,
+        cuda=True,
+        triton=True,
+        arg_types=None,
```
A type signature here would be appreciated, if you happen to know

ezyang commented:
Not landable as is, but the patch LGTM.

eellison commented on torch/_inductor/fx_passes/joint_graph.py:
```diff
@@ -194,6 +194,7 @@ def add_node_replacement(self, node: torch.fx.Node, tensor: torch.Tensor) -> Non
 
 @torch.utils._python_dispatch._disable_current_modes()
 def constant_fold_uniform_value(gm: torch.fx.GraphModule):
+    return
```
Do you have a repro of succeeds without constant folding and fails with it?

eellison replied:
Nevermind found it - will submit fix

desertfire committed:
[AOTI] Fix a cpp kernel missing arg type issue (#118353)

Summary:

The current way of fetching the kernel arg types only works for tensors, not symbols.

Test Plan: CI

Differential Revision: D53111282
